packages/pie/src/PieLayout.js

```diff
@@ -94,103 +22,145 @@ export const PieLayoutDefaultProps = {
                 const angle = Math.abs(arc.endAngle - arc.startAngle)
 
                 return {
                     ...arc,
                     angle,
                     angleDeg: radiansToDegrees(angle),
+                    color: getColor(arc.data),
                 }
             }),
-        })),
-        withPropsOnChange(['arcs', 'getColor'], ({ arcs, getColor }) => ({
-            arcs: arcs.map(arc => ({
-                ...arc,
-                color: getColor(arc.data),
-            })),
-        })),
-        pure
-    )(Component)
-
-export default setDisplayName('PieLayout')(enhance(PieLayout))
+
+        [pie, data, getColor]
+    )
+
+    const computedProps = React.useMemo(() => {
+        let radius = Math.min(width, height) / 2
+        let innerRadius = radius * Math.min(_innerRadius, 1)
+
+        let centerX = width / 2
+        let centerY = height / 2
+
+        let boundingBox
+        if (fit === true) {
+            const { points, ...box } = computeArcBoundingBox(
+                centerX,
+                centerY,
+                radius,
+                startAngle - 90,
+                endAngle - 90
+            )
+            const ratio = Math.min(width / box.width, height / box.height)
+
+            const adjustedBox = {
+                width: box.width * ratio,
+                height: box.height * ratio,
+            }
+            adjustedBox.x = (width - adjustedBox.width) / 2
+            adjustedBox.y = (height - adjustedBox.height) / 2
+
+            centerX = ((centerX - box.x) / box.width) * box.width * ratio + adjustedBox.x
+            centerY = ((centerY - box.y) / box.height) * box.height * ratio + adjustedBox.y
+
+            boundingBox = { box, ratio, points }
+
+            radius = radius * ratio
+            innerRadius = innerRadius * ratio
+        }
+
+        const arcGenerator = d3Arc()
+            .outerRadius(radius)
+            .innerRadius(innerRadius)
+            .cornerRadius(cornerRadius)
+
+        return {
+            centerX,
+            centerY,
+            radius,
+            innerRadius,
+            arcGenerator,
+            debug: boundingBox,
+        }
+    }, [width, height, _innerRadius, startAngle, endAngle, fit, cornerRadius])
+
+    return render({
+        arcs,
+        startAngle,
+        endAngle,
+        width,
+        height,
+        debug,
+        ...computedProps,
+    })
+}
+
+PieLayout.propTypes = {
+    data: PropTypes.arrayOf(
+        PropTypes.shape({
+            id: PropTypes.string.isRequired,
+            value: PropTypes.number.isRequired,
+        })
+    ).isRequired,
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+    fit: PropTypes.bool.isRequired,
+    sortByValue: PropTypes.bool.isRequired,
+    startAngle: PropTypes.number.isRequired,
+    endAngle: PropTypes.number.isRequired,
+    padAngle: PropTypes.number.isRequired,
+    // arcs: PropTypes.array.isRequired, // computed
```

Comment:
What was the reason for commenting these prop types out?

Reply:
All of the props with the `// computed` comment, from my understanding, were computed using other props.  `Arcs`, for example, was always recomputed if any one of the `pie`, `data`, or `getColor` props changed.  

I commented them out because they no longer need to be pipe-lined in as props and can just be computed using `React.useMemo` in the body of the function.  However, your question made me realize something.  Is there some sort of override feature that would be removed as a result of removing these props?  Previously, I think it wasn't possible to override since the `// computed` props were re-computed whenever the props previously mentioned were changed, but I'm not 100% sure.  

Also, when I double checked, one of the props,`innerRadius`, I commented out was `// re-computed` so I un-commented that prop since it should actually be passed in by the user.